### DISCUSSION TITLE
Remotely configured translations?

### DISCUSSION THREAD

**Discussion by BrutalCoding** [slang-i18n/slang#101] [Ideas]
*edit* - I just noticed this https://github.com/Tienisto/slang/issues/52 about the same issue. Only noticed it after I opened my topic here.

For anyone reading, what is the best approach to make (certain) translations remotely configurable?

Let me explain.

I've currently got 1 JSON with translations, and Slang generates the statically typed version of it.
So far so good.

However, some of these translations might change in the future. The keys will stay the same, but the values might change and that could also affect the variables inside the values. No new variables will be added but some might be removed/unused. 

E.g. simple use case
**Today:**
`"greeting": "Hey $username, how can I help you?"`
**In the future, without app updates:** 
`"greeting": "How may I help you today?"`

A more realistic use case is that a certain text is explaining how the (dollar) amount that the user is about to pay, is calculated. The user sees this before the user confirms the payment. This calculation is done by the backend, thus the amount might change over time and I would like to have the flexibility to update the explainer text to reflect this change too.

My solution would be to:
1. Keep Slang for all static translations that cant be changed remotely. I'll have all the benefits from Slang such as using dynamic variables, plurals, statically typed translations, context etc.
2. Have my custom implementation for the dynamic translations e.g. upon loading app, fetch english translation JSON from remote config and use that. It does mean I'll have to implement missing features myself e.g. dynamic variables ($username) and any others if I need them.

The way I see it is that I should implement my own approach for these remotely configurable translations, so Slang will have nothing to do with this topic. Because I don't see how Slang could help me in this scenario, but I'm obviously open to any ideas.

**Comment by Tienisto** [comment]
Hi,
Thanks for your input. I know that this feature is quite desired :)
It does work for simple use cases, but your flutter code may already have an argument (e.g. `t.greeting(name: myName)`).
Hence, we need a function and not a string. Slang could just ignore the `name` parameter in this case and return the string from your backend but I think this is not a good design :joy:.

This feature is on the pipeline but it is too much effort to do it now. Maybe, end of 2022 is possible.

**Comment by Tienisto** [comment]
This is released in 3.0.0
